builtin/providers/aws/resource_aws_iam_group.go

```diff
@@ -60,9 +58,10 @@ func resourceAwsIamGroupCreate(d *schema.ResourceData, meta interface{}) error {
 
 func resourceAwsIamGroupRead(d *schema.ResourceData, meta interface{}) error {
 	iamconn := meta.(*AWSClient).iamconn
+	name := d.Get("name").(string)
 
 	request := &iam.GetGroupInput{
-		GroupName: aws.String(d.Id()),
+		GroupName: aws.String(name),
```

Comment:
This change seems superfluous; `name` isn't used outside of this `iam.GetGroupInput` struct. 
Does it serve some purpose I don't see?